src/listen/listener-registry.js

```diff
@@ -842,23 +853,42 @@ module.exports = class ListenerRegistry {
   }
 
   /**
-  * Validates that the pattern is not empty and is a valid regular expression
+  * Converts a pattern to a RegExp object
   *
-  * @param   {SocketWrapper} socketWrapper
-  * @param   {String} pattern
+  * Expects the pattern to be a glob expression unless the `enableRegexListenPatterns` option is
+  * set, in which case it expects a valid regular expression
+  *
+  * returns false if the pattern is empty or is not a valid glob/regular expression
+  *
+  * @param   {SocketWrapper?} socketWrapper
+  * @param   {String}         pattern
   *
   * @returns {RegExp}
   */
-  _validatePattern (socketWrapper, pattern) {
+  _compilePattern (socketWrapper, pattern) {
     if (!pattern) {
       return false
     }
-
-    try {
-      return new RegExp(pattern)
```

Comment:
there's no ambiguity, if `enableRegexListenPatterns` is set, patterns will be parsed as regular expressions, otherwise they will be parsed as globs